### PR TITLE
improving selectors

### PR DESCRIPTION
**Pull request by JulianO911**

### CODE REVIEW

File: cypress/page/menu-content.page.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ class MenuContentPage {
   private menuContentPageURL: string
 
   constructor() {
-      this.tShirtMenu = "#block_top_menu > ul > li:nth-child(3) > a";
+      /* Now the selector searchs the last li in Ul and and a label
+        that is inside the li */
+      this.tShirtMenu = "#block_top_menu > ul > li:last-child a";
```
Comment (holgiosalos) [inline cypress/page/menu-content.page.ts]: according to what we talked in last class, this selector can use the title attribute of the `a` tag

File: cypress/page/products-list.page.ts [hunk 2]
```diff
@@ -3,8 +3,10 @@ class ProductsList {
     private proceedToCheckoutButton: string;
 
     constructor(){
-        this.addToChartButton = "#center_column a.button.ajax_add_to_cart_button.btn.btn-default";
-        this.proceedToCheckoutButton = "[style*='display: block;'] .button-container > a";
+        // Just get a span that's inside in a ajax_add_to_cart class button.
```
Comment (holgiosalos) [inline cypress/page/products-list.page.ts]: the justification is only required when you are not agree with one suggestion, please remove the comments 😉

File: cypress/page/products-list.page.ts [hunk 3]
```diff
@@ -3,8 +3,10 @@ class ProductsList {
...
+        this.addToChartButton = ".ajax_add_to_cart_button span";
+        // Get the a label that's inside in a label with layer_cart id.
+        this.proceedToCheckoutButton = "#layer_cart a";
```
Comment (holgiosalos) [inline cypress/page/products-list.page.ts]: for these two you can use an attribute to specify that we are looking for the add to cart and the proceed to checkout button

File: cypress/page/shoping-cart.page.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ class ShopingCart {
     private proceedToCheckoutButton: string;
 
     constructor(){
+        // There's no need to improve this selector.
         this.proceedToCheckoutButton =  ".cart_navigation span"
```
Comment (holgiosalos) [inline cypress/page/shoping-cart.page.ts]: yes, you can improve it, try to use attributes to indicate we are looking for the proceed to checkout button

**Review by holgiosalos — COMMENTED** [review]
Selectors on the following page objects can be improved:

1. address-step.page.ts
2. Proceed to checkout button of shipping-step.page.ts

File: cypress/page/products-list.page.ts (hunk 2 above)
Comment (JulianO911) [reply cypress/page/products-list.page.ts]: I understood that the selectors should work even if software got changes. The tittle attribute is susceptible to changes right? That means if the tittle changes then the selector won't work...

Comment (holgiosalos) [reply cypress/page/products-list.page.ts]: tag attributes are less likely to change over time than CSS classes, also using attributes improves the readability of the code.

File: cypress/page/menu-content.page.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@ class MenuContentPage {
   constructor() {
       /* Now the selector searchs the last li in Ul and and a label
         that is inside the li */
-      this.tShirtMenu = "#block_top_menu > ul > li:last-child a";
+      this.tShirtMenu = "#block_top_menu > ul > li:last-child a[title='T-shirts']";
```
Comment (holgiosalos) [inline cypress/page/menu-content.page.ts]: remove the `:last-child` -> `"#block_top_menu > ul > li > a[title='T-shirts']"` should be fine

File: cypress/page/products-list.page.ts [hunk 6]
```diff
@@ -4,9 +4,9 @@ class ProductsList {
 
     constructor(){
         // Just get a span that's inside in a ajax_add_to_cart class button.
-        this.addToChartButton = ".ajax_add_to_cart_button span";
+        this.addToChartButton = ".ajax_add_to_cart_button[title='Add to cart']";
```
Comment (holgiosalos) [inline cypress/page/products-list.page.ts]: just `[title='Add to cart']` should work

File: cypress/page/shoping-cart.page.ts [hunk 7]
```diff
@@ -3,7 +3,7 @@ class ShopingCart {
 
     constructor(){
         // There's no need to improve this selector.
-        this.proceedToCheckoutButton =  ".cart_navigation span"
+        this.proceedToCheckoutButton =  ".cart_navigation a[title='Proceed to checkout']"
```
Comment (holgiosalos) [inline cypress/page/shoping-cart.page.ts:5]: just `a[title='Proceed to checkout']` should work

**Review by holgiosalos — COMMENTED** [review]
remember to remove all the comments related to the selectors

File: cypress/page/shoping-cart.page.ts (hunk 7 above)
Comment (JulianO911) [reply cypress/page/shoping-cart.page.ts:5]: that selector matchs with two results, so i couldn't change it :/

Comment (JulianO911) [reply cypress/page/shoping-cart.page.ts:5]: image

Comment (holgiosalos) [reply cypress/page/shoping-cart.page.ts:5]: ok

**Review by holgiosalos — COMMENTED** [review]
Remember that selectors on the following page objects can be improved:

1. address-step.page.ts
2. Proceed to checkout button of shipping-step.page.ts

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**